modules/apps/analytics/analytics-client-js/src/plugins/visibility.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const applicationId = 'Page';
+
+/**
+ * Handle differents browser versions to Visibility API
+ */
+function getHiddenKey() {
+	let hidden, visibilityChange;
+
+	if (typeof document.hidden !== 'undefined') {
+		hidden = 'hidden';
+		visibilityChange = 'visibilitychange';
+	}
+	else if (typeof document.msHidden !== 'undefined') {
+		hidden = 'msHidden';
+		visibilityChange = 'msvisibilitychange';
+	}
+	else if (typeof document.webkitHidden !== 'undefined') {
+		hidden = 'webkitHidden';
+		visibilityChange = 'webkitvisibilitychange';
+	}
+
+	return {
+		hidden,
+		visibilityChange,
+	};
+}
+
+/**
+ * Sends tabBlurred or tabFocused event on visibilityChange
+ * @param {Object} analytics The Analytics client
+ */
+function handleVisibilityChange(analytics) {
+	const {hidden} = getHiddenKey();
+
+	if (document[hidden]) {
+		analytics.send('tabBlurred', applicationId);
+	}
+	else {
+		analytics.send('tabFocused', applicationId);
+	}
+}
+
+/**
+ * Plugin function that registers listeners against browser visibility tabs
+ * @param {Object} analytics The Analytics client
+ */
+function visibility(analytics) {
+	const {visibilityChange} = getHiddenKey();
+	const onVisibilityChange = handleVisibilityChange.bind(null, analytics);
+
+	document.addEventListener(visibilityChange, onVisibilityChange);
```

Comment:
We should wrap this in an `if` statement so that we don't add the event listeners in the case that `visibilityChange` doesn't exist.